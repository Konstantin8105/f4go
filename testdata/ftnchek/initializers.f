C  Testing syntax recognition and assignment type checking of
C  standard and non-standard initializing declarations.

C  Here we use modern standard F90 initializers
      subroutine f90
      integer :: m, n = 100
C ftnchek does not recognize the following one yet
      integer, dimension(5) :: a = (/ 1, 2, 3, 4, 5 /)
      real :: pi = 3.14159265358979d0
      character :: c = 32   ! type mismatch
      integer :: q = 'hello'   ! type mismatch
      print *, 'F90 initializers:'
      print *, m, n, a   ! m used before set
      print *, pi, c, q
      end
C  Here we use archaic but standard F77 separate type decls and data stmts
      subroutine f77
      integer m, n, a(5)
      real pi
      character c
      integer q
      data n / 100 /
      data a /  1, 2, 3, 4, 5 /
      data pi / 3.14159265358979d0 /
      data c / 32 /   ! type mismatch
      data q / 'hello' /   ! type mismatch
      print *, 'F77 initializers:'
      print *, m, n, a   ! m used before set
      print *, pi, c, q
      end
C  This one uses "bastard" initializer form.
      subroutine bastard
      integer m, n / 100 /, a(5) / 1, 2, 3, 4, 5 /
      real pi / 3.14159265358979d0 /
      character c / 32 /   ! type mismatch
      integer q / 'hello' /   ! type mismatch
      print *, 'bastard initializers:'
      print *, m, n, a   ! m used before set
      print *, pi, c, q
      end
C  Main program just suppresses "never invoked" warnings.
      program main
      call f90
      call f77
      call bastard
      end

