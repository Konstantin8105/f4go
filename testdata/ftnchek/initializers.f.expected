COMMENT             	C  Testing syntax recognition and assignment type checking of
NEW_LINE            	

COMMENT             	C  standard and non-standard initializing declarations.
NEW_LINE            	

NEW_LINE            	

COMMENT             	C  Here we use modern standard F90 initializers
NEW_LINE            	

SUBROUTINE          	subroutine
IDENT               	f90
NEW_LINE            	

INTEGER             	integer
DOUBLE_COLON        	::
IDENT               	m
,                   	,
IDENT               	n
=                   	=
INT                 	100
NEW_LINE            	

COMMENT             	C ftnchek does not recognize the following one yet
NEW_LINE            	

INTEGER             	integer
,                   	,
IDENT               	dimension
(                   	(
INT                 	5
)                   	)
DOUBLE_COLON        	::
IDENT               	a
=                   	=
(                   	(
/                   	/
INT                 	1
,                   	,
INT                 	2
,                   	,
INT                 	3
,                   	,
INT                 	4
,                   	,
INT                 	5
/                   	/
)                   	)
NEW_LINE            	

IDENT               	real
DOUBLE_COLON        	::
IDENT               	pi
=                   	=
FLOAT               	3.14159265358979d0
NEW_LINE            	

IDENT               	character
DOUBLE_COLON        	::
IDENT               	c
=                   	=
INT                 	32
COMMENT             	! type mismatch
NEW_LINE            	

INTEGER             	integer
DOUBLE_COLON        	::
IDENT               	q
=                   	=
STRING              	'hello'
COMMENT             	! type mismatch
NEW_LINE            	

IDENT               	print
*                   	*
,                   	,
STRING              	'F90 initializers:'
NEW_LINE            	

IDENT               	print
*                   	*
,                   	,
IDENT               	m
,                   	,
IDENT               	n
,                   	,
IDENT               	a
COMMENT             	! m used before set
NEW_LINE            	

IDENT               	print
*                   	*
,                   	,
IDENT               	pi
,                   	,
IDENT               	c
,                   	,
IDENT               	q
NEW_LINE            	

END                 	end
NEW_LINE            	

COMMENT             	C  Here we use archaic but standard F77 separate type decls and data stmts
NEW_LINE            	

SUBROUTINE          	subroutine
IDENT               	f77
NEW_LINE            	

INTEGER             	integer
IDENT               	m
,                   	,
IDENT               	n
,                   	,
IDENT               	a
(                   	(
INT                 	5
)                   	)
NEW_LINE            	

IDENT               	real
IDENT               	pi
NEW_LINE            	

IDENT               	character
IDENT               	c
NEW_LINE            	

INTEGER             	integer
IDENT               	q
NEW_LINE            	

IDENT               	data
IDENT               	n
/                   	/
INT                 	100
/                   	/
NEW_LINE            	

IDENT               	data
IDENT               	a
/                   	/
INT                 	1
,                   	,
INT                 	2
,                   	,
INT                 	3
,                   	,
INT                 	4
,                   	,
INT                 	5
/                   	/
NEW_LINE            	

IDENT               	data
IDENT               	pi
/                   	/
FLOAT               	3.14159265358979d0
/                   	/
NEW_LINE            	

IDENT               	data
IDENT               	c
/                   	/
INT                 	32
/                   	/
COMMENT             	! type mismatch
NEW_LINE            	

IDENT               	data
IDENT               	q
/                   	/
STRING              	'hello'
/                   	/
COMMENT             	! type mismatch
NEW_LINE            	

IDENT               	print
*                   	*
,                   	,
STRING              	'F77 initializers:'
NEW_LINE            	

IDENT               	print
*                   	*
,                   	,
IDENT               	m
,                   	,
IDENT               	n
,                   	,
IDENT               	a
COMMENT             	! m used before set
NEW_LINE            	

IDENT               	print
*                   	*
,                   	,
IDENT               	pi
,                   	,
IDENT               	c
,                   	,
IDENT               	q
NEW_LINE            	

END                 	end
NEW_LINE            	

COMMENT             	C  This one uses "bastard" initializer form.
NEW_LINE            	

SUBROUTINE          	subroutine
IDENT               	bastard
NEW_LINE            	

INTEGER             	integer
IDENT               	m
,                   	,
IDENT               	n
/                   	/
INT                 	100
/                   	/
,                   	,
IDENT               	a
(                   	(
INT                 	5
)                   	)
/                   	/
INT                 	1
,                   	,
INT                 	2
,                   	,
INT                 	3
,                   	,
INT                 	4
,                   	,
INT                 	5
/                   	/
NEW_LINE            	

IDENT               	real
IDENT               	pi
/                   	/
FLOAT               	3.14159265358979d0
/                   	/
NEW_LINE            	

IDENT               	character
IDENT               	c
/                   	/
INT                 	32
/                   	/
COMMENT             	! type mismatch
NEW_LINE            	

INTEGER             	integer
IDENT               	q
/                   	/
STRING              	'hello'
/                   	/
COMMENT             	! type mismatch
NEW_LINE            	

IDENT               	print
*                   	*
,                   	,
STRING              	'bastard initializers:'
NEW_LINE            	

IDENT               	print
*                   	*
,                   	,
IDENT               	m
,                   	,
IDENT               	n
,                   	,
IDENT               	a
COMMENT             	! m used before set
NEW_LINE            	

IDENT               	print
*                   	*
,                   	,
IDENT               	pi
,                   	,
IDENT               	c
,                   	,
IDENT               	q
NEW_LINE            	

END                 	end
NEW_LINE            	

COMMENT             	C  Main program just suppresses "never invoked" warnings.
NEW_LINE            	

PROGRAM             	program
IDENT               	main
NEW_LINE            	

CALL                	call
IDENT               	f90
NEW_LINE            	

CALL                	call
IDENT               	f77
NEW_LINE            	

CALL                	call
IDENT               	bastard
NEW_LINE            	

END                 	end
NEW_LINE            	

NEW_LINE            	

